src/d3a/d3a_core/simulation.py

```diff
@@ -241,6 +241,7 @@ def _run_cli_execute_cycle(self, slot_resume, tick_resume):
 
     def _update_and_send_results(self, is_final=False):
         self.endpoint_buffer.update_stats(self.area, self.status, self.progress_info)
+        self.endpoint_buffer.send_results_to_child_areas(self.area)
```

Comment:
Why not calling this method in the endpoint_buffer.update_area_aggregated_stats ? It serves the same purpose, and you can avoid the second recursion.

Reply:
Nice! Didnt see that. Will add it there. Thanks!